sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2405,22 +2496,34 @@ static ur_result_t SetKernelParamsAndLaunch(
                               : Empty);
   }
 
+  std::vector<ur_exp_kernel_arg_properties_t> UrArgs;
```

Comment:
Would it be possible to keep this vector inside of the queue object so that we don't allocate it (potentially multiple times if there's a lot of arguments) for every kernel launch?
We have a similar mechanism in L0v2 adapter here:
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/v2/command_list_manager.hpp#L177